modules/collect/src/test/java/com/opengamma/strata/collect/result/FailureItemTest.java

```diff
@@ -75,4 +75,15 @@ public void test_of_reasonMessageExceptionNestedException() {
     assertEquals(test.toString(), "INVALID: my big bad failure: java.lang.IllegalArgumentException: message");
   }
 
+  public void test_of_reasonMessageExceptionNestedExceptionWithAttributes() {
+    IllegalArgumentException innerEx = new IllegalArgumentException("inner");
+    IllegalArgumentException ex = new IllegalArgumentException("message", innerEx);
+    FailureItem test = FailureItem.of(FailureReason.INVALID, ex, "my {foo} {bar} failure", "big", "bad");
+    assertEquals(test.getReason(), FailureReason.INVALID);
+    assertEquals(test.getMessage(), "my big bad failure");
+    assertEquals(test.getCauseType().isPresent(), true);
+    assertEquals(test.getCauseType().get(), IllegalArgumentException.class);
+    assertEquals(test.getStackTrace().contains(".test_of_reasonMessageExceptionNestedExceptionWithAttributes("), true);
+    assertEquals(test.toString(), "INVALID: my big bad failure: java.lang.IllegalArgumentException: message : {bar=bad, foo=big}");
```

Comment:
It would be good to test the attributes map directly rather than doing it via the `toString()` method.